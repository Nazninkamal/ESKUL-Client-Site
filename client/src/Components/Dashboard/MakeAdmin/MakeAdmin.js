import React from 'react';
import axios from 'axios';
import { Button, TextField } from '@mui/material';
import { Box } from '@mui/system';
import { useForm } from "react-hook-form";
import { api } from '../../../Hooks/Api';
import { alert, ButtonStyle } from '../../../Hooks/useStyle';
const MakeAdmin = () => {
    const { register, handleSubmit, formState: { errors }, reset } = useForm();
    const onSubmit = data => {
        axios.patch(`${api}/users/admin`, data)
            .then((response) => {
                response.status === 200 &&
                    alert('success', 'Make a admin Successfully')
            })
            .catch((error) => {
                !error.status === 200 &&
                    alert('error', 'Bad Request, Places Try again')
                console.log(error);
            });


        reset();
    };


    return (
        <>
            <h1>Make a admin form</h1>
            <Box sx={{ maxWidth: 400 }}>
                <form onSubmit={handleSubmit(onSubmit)}>
                    <TextField size='small' fullWidth label="Email" placeholder='Email'{...register("email", { required: true })} />
                    {errors.email && <span>This admin field is required</span>} <br /> <br />
                    <Button type='submit' fullWidth sx={ButtonStyle}>Make Admin</Button>
                </form>
            </Box>
        </>
    );
};

export default MakeAdmin;