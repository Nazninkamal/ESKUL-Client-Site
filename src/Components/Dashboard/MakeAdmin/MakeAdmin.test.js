
import { shallow } from 'enzyme';
import MakeAdmin from './MakeAdmin';


    describe('Make Admin testing', () => {
      it('should render correctly in "debug" mode', () => {
        const component = shallow(<MakeAdmin/>);
        expect(component).toMatchSnapshot();
      });
 
    it('text check', () =>{
      let wrapper = shallow(<MakeAdmin/>)
      expect(wrapper.contains("Make an admin form")).toEqual(true)
   })

 
  
})