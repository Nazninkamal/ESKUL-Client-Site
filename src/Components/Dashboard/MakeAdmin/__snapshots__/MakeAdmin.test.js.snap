// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Admin testing should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MakeAdmin should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MakeAdmin test should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
